src/Dialogs/Cloud/WeGlideUploadDialog.cpp

```diff
@@ -0,0 +1,191 @@
+    // StaticString<0x40> pilot_name;
+    StaticString<0x40> scoring_date;
+    StaticString<0x40> registration;
+    StaticString<0x40> competition_id;
+  } UploadResponse;
+
+  static Co::InvokeTask UpdateTask(Path igc_path, const char* uri,
+    const WeGlide::Pilot& pilot,
+    uint_least32_t glider_id,
+    UploadResponse& response,
+    ProgressListener& progress) noexcept {
+    boost::json::value json =
+      co_await WeGlide::UploadFlight(*Net::curl, uri, pilot, glider_id,
+        igc_path, progress);
+    auto json_value = json.get_array().at(0);
+    response.scoring_date = UTF8ToWideConverter(
+      Json::Lookup(json_value, "scoring_date")->get_string().c_str());
+    response.flight_id = Json::Lookup(json_value, "id")->get_int64();
+    response.registration = UTF8ToWideConverter(
+      Json::Lookup(json_value, "registration")->get_string().c_str());
+    response.competition_id = UTF8ToWideConverter(
+      Json::Lookup(json_value, "competition_id")->get_string().c_str());
+    auto user = Json::Lookup(json_value, "user")->as_object();
+
+    response.pilot.id =
+      Json::Lookup(user, "id")->get_int64();
+    response.pilot.name = UTF8ToWideConverter(
+      Json::Lookup(user, "name")->get_string().c_str());
+
+    auto aircraft = Json::Lookup(json_value, "aircraft")->as_object();
+
+    response.glider.id = Json::Lookup(aircraft, "id")->get_int64();
+    response.glider.name = UTF8ToWideConverter(Json::Lookup(aircraft, "name")->get_string().c_str());
+    response.glider.kind = UTF8ToWideConverter(Json::Lookup(aircraft, "kind")->get_string().c_str());
+    response.glider.sc_class = UTF8ToWideConverter(Json::Lookup(aircraft, "sc_class")->get_string().c_str());
+
+  }
+
+  bool UploadFlightDialog(Path igc_path, WeGlide::Pilot pilot, uint_least32_t glider_id) {
+    const WeGlideSettings& weglide_settings =
+      CommonInterface::GetComputerSettings().weglide;
+    StaticString<0x1000> msg;
+    NarrowString<0x200> url(weglide_settings.default_url);
+    url += "/igcfile";
+
+    try {
+      if (glider_id == 0)
+        glider_id = CommonInterface::GetComputerSettings().plane
+        .weglide_glider_type;
+      if (pilot.id == 0) {
+        pilot = weglide_settings.pilot;
+      }
+      UploadResponse response;
+      PluggableOperationEnvironment env;
+      LogFormat(_("WeGlide Upload: %s"), igc_path.c_str());
+      if (!File::Exists(LocalPath(igc_path))) {
+        msg.Format(_("Error! File '%s' doesn't exist!"), igc_path.c_str());
+        LogFormat(_("WeGlide Upload: %s"), msg.c_str());
+        ShowMessageBox(msg, _T("WeGlide Upload"), MB_OK | MB_ICONEXCLAMATION);
+
+        return false;
+      }
+
+      if (ShowCoDialog(UIGlobals::GetMainWindow(), UIGlobals::GetDialogLook(),
+        _("Download"), UpdateTask(igc_path, url, pilot,
+          glider_id, response, env), &env) == false)
+      {
+        return false;
+      }
+
+      msg.Format(_("File upload '%s' was succesfull!\n\n"), igc_path.c_str());
```

Comment:
No exclamation marks and no layout (newlines) in translatable strings.

Reply:
Is such an expression allowed?

`msg.Format(_("File upload '%s' was succesfull%s"), igc_path.c_str(), _T("!\n\n"));`

Reply:
No. Why do you want to have the exclamation mark? Why do you want to have the newlines?
How should translators know what this last "%s" is?
With your way of phrasing it, it isn't even obvious what the first one means. "Upload of '%s' was successful" would have been more obvious.

Reply:
Ok!
The newlines I need for the messagebox, where the string is displayed, but I can append this in an additional step too. I know that isn't the best way to display it, but a new dialog box with bettter and structured fields needs additional work too.

Reply:
BTW. in the lines below in the msg.AppendFormat() function I changed the '_(' to '_T(' because there are no translatable strings, alright?

Reply:
Of course, but I just wanted to hear your thoughts on it one more time ...